src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/BufferSegmentStack.cs

```diff
@@ -20,6 +20,10 @@ public BufferSegmentStack(int size)
             _size = 0;
         }
 
+        // We re-use the array as the lock guarding the state of the Pipe.
+        // this shouldn't be used anywhere else.
+        public object Sync => _array;
+
```

Comment:
Isn't this stack resizable? What happens when it grows and the array is replaced?

Reply:
Gahhhh, I missed this 😢

Reply:
OK I'll go with plan B...

Reply:
OK done...